lib/src/solver/package_lister.dart

```diff
@@ -49,8 +49,8 @@ class PackageLister {
   /// The type of the dependency from the root package onto [_ref].
   final DependencyType _dependencyType;
 
-  /// The set of package names that were overridden by the root package.
-  final Set<String> _overriddenPackages;
+  /// The set of packages that were overridden by the root package.
+  final Set<PackageRange> _overriddenPackages;
```

Comment:
Wouldn't it make sense to keep this as a `Set<String>`?

Reply:
Done